src/apiProxy/api-proxy.js

```diff
@@ -92,40 +113,48 @@ function urlFormatParams(params, doEncode) {
  *
  * @param {Object} externalRequestOpts request options that will be applied to
  *   every query request
- * @param {Object} apiConfig { endpoint, params }
+ * @param {Object} apiConfig { endpoint, params, flags }
  *   call)
  * @return {Object} externalRequestOptsQuery argument for the call to
  *   `externalRequest` for the query
  */
-export const buildRequestArgs = externalRequestOpts => ({ endpoint, params }) => {
-	const externalRequestOptsQuery = { ...externalRequestOpts };
-	externalRequestOptsQuery.url = `/${endpoint}`;
-
-	const dataParams = urlFormatParams(params, externalRequestOptsQuery.method === 'get');
-
-	switch (externalRequestOptsQuery.method) {
-	case 'get':
-		externalRequestOptsQuery.url += `?${dataParams}`;
-		externalRequestOptsQuery.headers['X-Meta-Photo-Host'] = 'secure';
-		break;
-	case 'post':
-		externalRequestOptsQuery.body = dataParams;
-		externalRequestOptsQuery.headers['content-type'] = 'application/x-www-form-urlencoded';
-		break;
-	}
+export const buildRequestArgs = externalRequestOpts =>
+	({ endpoint, params, flags }) => {
 
-	return externalRequestOptsQuery;
-};
+		// cheap, brute-force object clone, acceptable for serializable object
+		const opts = JSON.parse(JSON.stringify(externalRequestOpts));
+		opts.url = `/${endpoint}`;
+
+		const formattedParams = urlFormatParams(params, opts.method === 'get');
+
+		if (flags) {
+			opts.headers['X-Meetup-Request-Flags'] = flags.join(',');
+		}
```

Comment:
this is the only 'new' block of code in this function - the rest is mostly an indentation change